components/layout/block.rs

```diff
@@ -1196,24 +1196,19 @@ impl BlockFlow {
 
                 // Can't use `for` because we assign to
                 // `candidate_block_size_iterator.candidate_value`.
-                loop {
-                    match candidate_block_size_iterator.next() {
-                        Some(block_size_used_val) => {
-                            solution = Some(
-                            BSizeConstraintSolution::solve_vertical_constraints_abs_nonreplaced(
-                                    block_size_used_val,
-                                    margin_block_start,
-                                    margin_block_end,
-                                    block_start,
-                                    block_end,
-                                    content_block_size,
-                                    available_block_size));
-
-                            candidate_block_size_iterator.candidate_value
-                                = solution.unwrap().block_size;
-                        }
-                        None => break,
-                    }
+                while let Some(block_size_used_val) =  candidate_block_size_iterator.next() {
```

Comment:
@Manishearth in my current code, I've replaced:
 `while let Some(block_size_used_val) = candidate_block_size_iterator.next()` 
with:
`for block_size_used_val in candidate_block_size_iterator`

I'm unsure if this is correct. It compiles fine, however [this](https://github.com/servo/servo/blob/6ab205a97e0b19adbb28c9fd18e34faf81dde589/components/layout/block.rs#L1170-L1171) indicates that a for loop cannot be used in this instance.

An identical comment (for a different function but similar circumstance) appears [here](https://github.com/servo/servo/blob/6ab205a97e0b19adbb28c9fd18e34faf81dde589/components/layout/block.rs#L924-925)